src/components/DownloadExperiment.js

```diff
@@ -0,0 +1,187 @@
+import React from 'react';
+
+import Button from './Button';
+import ProcessingGears from './ProcessingGears';
+import ModalManager from './Modal/ModalManager';
+import {
+  DatasetDownloadOptionsForm,
+  AggreationOptions,
+  TransformationOptions,
+  AdvancedOptions,
+  EmailAddressOptions,
+  SubmitDatasetOptionsForm,
+} from './DatasetDownloadOptionsForm';
+
+import { getDataSet } from '../api/dataSet';
+
+import { useLocalStorage } from '../common/hooks';
+import { getDomain } from '../common/helpers';
+
+const DownloadExperiment = ({ experiment, showProcessing = false }) => {
+  const [emailAddress] = useLocalStorage('email-address');
+
+  const {
+    accession_code: experimentAccessionCode,
+    num_downloadable_samples: downloadableSamplesCount,
+  } = experiment;
+
+  const data = {};
+  data[experimentAccessionCode] = experiment.samples.map(s => s.accession_code);
+
+  const emptyDataset = {
+    data,
+    aggregate_by: 'EXPERIMENT',
+    quantile_normalize: true,
+    scale_by: 'NONE',
+    email_address: undefined,
+    email_ccdl_ok: false,
+  };
+
+  const [dataset, setDataset] = React.useState(emptyDataset);
+
+  const [processingDataset, setDatasetIsProcessing] = useLocalStorage(
+    `dataset-${experiment.accession_code}`,
+    false
+  );
+
+  // check if dataset is being processed
+  // if not clean up localstorage
+  React.useEffect(() => {
+    const datasetId = dataset.id || processingDataset;
+    const refreshState = async () => {
+      const refreshedDataset = await getDataSet(datasetId);
+
+      if (refreshedDataset.is_processing) {
+        setDataset(refreshedDataset);
+        setDatasetIsProcessing(refreshedDataset.id);
+      } else {
+        setDataset(emptyDataset);
+        setDatasetIsProcessing(null);
+      }
+    };
+
+    const justStarted = dataset.id !== processingDataset;
+    const wasProccessing = processingDataset && !dataset.id;
+
+    if (datasetId && (justStarted || wasProccessing)) refreshState();
+  }, [dataset, setDataset, processingDataset, setDatasetIsProcessing]);
+
+  // Hide this component
+  if (downloadableSamplesCount === 0) return false;
```

Comment:
Should this be the responsability of the parent? To only show the component when there're downloadable samples?

Reply:
yeah that would be a good call for this case (actually I already think it's implemented for the case when there are no downloadable samples, but I think we should still handle the case where we pass a parameter to hide when the dataset is being processed and the button shouldn't show, Just to keep the logic more dry...